### PR TITLE
Homepage redesign

### PR DESCRIPTION
**Pull request by LizBaker**
Closes https://github.com/newrelic/docs-website/issues/4431

### Desktop
Screen Shot 2021-11-16 at 2 57 09 PM
Screen Shot 2021-11-16 at 2 57 48 PM

### Mobile
Screen Shot 2021-11-16 at 2 58 14 PM
Screen Shot 2021-11-16 at 2 58 24 PM

### CODE REVIEW

**Comment by legnaleama** [comment]
It's looking good @LizBaker !

File: src/components/HomepageBanner.js
Comment (clarkmcadoo) [inline src/components/HomepageBanner.js:15]: massive nit-pick, but could we set the calculation numbers as variables somewhere? maybe at the top of this file? That way it's easier to tell where the numbers came from. **Disclosure** I can def be talked out of this if this would be a huge pain and majorly divergent to the current pattern

```suggestion
        max-height: ${MAX_HEIGHT_DESKTOP}
        padding-top: calc(${MAX_HEIGHT_DESKTOP} / ${MAX_WIDTH_DESKTOP} * 100%);
```

File: src/components/HomepageBanner.js
Comment (clarkmcadoo) [inline src/components/HomepageBanner.js:175]: We might look at making the buttons the same width < 900px width, but I'll def leave that up to @legnaleama

File: src/components/HomepageBanner.js
Comment (clarkmcadoo) [inline src/components/HomepageBanner.js:6]:
```suggestion
const MAX_HEIGHT_DESKTOP = 415
const MAX_WIDTH_DESKTOP = 1252
const TABLET_BREAKPOINT = 900
```

Comment (clarkmcadoo) [reply src/components/HomepageBanner.js:6]: just did those 3 as an example, but we'd set all of them here

File: src/components/HomepageBanner.js
Comment (clarkmcadoo) [reply src/components/HomepageBanner.js:175]: Just double-checked the designs, disregard this!

**Comment by roadlittledawn** [comment]
i believe they want all sections from New Relic Integrations and down removed entirely. waiting for confirmation from Betsy

File: src/components/HomepageBanner.js
Comment (LizBaker) [reply src/components/HomepageBanner.js:15]: we definitely can, this is actually a weird hack to make the div the banner lives in respect a certain aspect ratio. so more like MAX_HEIGHT_BANNER/MAX_WIDTH_DESKTOP

File: src/pages/index.js
Comment (zstix) [inline src/pages/index.js:37]: If we only need three, then we can add a `limit` to the graphql query, which should speed up the time it takes to build this page. I'll comment a suggestion down below too.

```suggestion
  const latestWhatsNewPosts = whatsNewPosts.map((edge) => {
```

File: src/pages/index.js
Comment (zstix) [inline src/pages/index.js:254]: I mentioned it above in this file: if we only need three posts, let's limit the query. That should speed this up a little bit.

```suggestion
      sort: {
        fields: [frontmatter___releaseDate, frontmatter___title]
        order: [DESC, ASC]
      }
      filter: { fields: { slug: { regex: "/whats-new/" } } }
      limit: 3
```

File: src/pages/index.js
Comment (zstix) [inline src/pages/index.js:261]: I don't actually think that we are using the release date on this page. We can probably remove it from the query (and the related propTypes).

```suggestion
```

**Review by zstix — COMMENTED** [review]
This looks so much better!!! That said, I had a few comments that we might want to consider incorporating.

Also, one other small nitpick: I found that the integrations looked a little weird without a border. What do you think about us adding a 'lil something to help improve them feeling like "tiles":

Screen Shot 2021-11-16 at 17 05 53

**Comment by legnaleama** [comment]
Hey @LizBaker, this is looking great, just a few amends: 

### Banner
The banner buttons aren't the same as in the Figma, both in style and hover effects. I'm aware I'm introducing some refactoring tasks here but I'd like to know what the long term implementation would be for those changes and/or if we could fit any at all in this PR. 

**Main differences**
- On light mode: both CTAs are using the dark mode version
- On both light and dark modes: Hover effect doesn't match the prototype in Figma

### CTAs
There seems to be a weird scaling going on, is this part of some Gatsby library? Ideally we should re-use the default button sizes (ie: SM, MD, LG) and tailor them to each breakpoint. In the past I've also used variants of those 3 scales for mobile and desktop independently but I think that is a bit overkill.
https://user-images.githubusercontent.com/3636549/142199005-54da63eb-86ff-4b3a-b54c-379c2e5f8de1.mov


### Mobile
**Layout issues on the banner**
- On mobile: The order of the buttons is reversed. Currently the secondary CTA ("Find your quickstart") is the first button. On the original design the free account CTA should be the main first action.

**Banner details**
- This is probably my fault but I noticed we use the same foreground image for desktop and mobile. I just went ahead an isolated the mobile version on Figma so it's easier to export. The previous one was also the wrong color on my side, sry.
- I also added a cropped version of the photo for mobile for better alignment with the decoration elements.
image

**Review by roadlittledawn — DISMISSED** [review]
we can remove all sections from `New Relic integrations` on down

File: src/pages/index.js
Comment (LizBaker) [reply src/pages/index.js:37]: oh perfect!

File: src/pages/index.js
Comment (LizBaker) [reply src/pages/index.js:261]: left behind when i pasted the query 😅

File: src/pages/index.js
Comment (LizBaker) [reply src/pages/index.js:136]: ah nope, oversight

File: src/components/HomepageBanner.js
Comment (LizBaker) [reply src/components/HomepageBanner.js:187]: just gave it that variant (it's required anyway) but found the colors don't match the design so i'm overriding them for now

File: src/components/HomepageBanner.js
Comment (LizBaker) [reply src/components/HomepageBanner.js:157]: good idea! i was also considering using emotion's styled function but i'm not sure which is better

File: src/components/HomepageBanner.js
Comment (LizBaker) [reply src/components/HomepageBanner.js]: gotcha, i think the sliver of pixels is now fixed with a new svg!

File: src/components/HomepageBanner.js
Comment (LizBaker) [reply src/components/HomepageBanner.js:183]: very true, i started with separate elements and it became weirder from there, gonna see how a refactor might look

**Comment by austin-schaefer** [comment]
Looking great @LizBaker ! One bit of feedback from me: Can we decrease the header size on `Welcome` and `Getting started`/`What's new`/`Popular content` sections? They look a bit too large to me relative to the rest of the copy and headers. 

image

**Comment by austin-schaefer** [comment]
cc @legnaleama on my above comment re:header size

**Comment by LizBaker** [comment]
@austin-schaefer definitely game if @legnaleama is!

File: src/components/HomepageBanner.js
Comment (zstix) [reply src/components/HomepageBanner.js]: FWIW, I'm still seeing the 'lil sliver with the new image:

Screen Shot 2021-11-18 at 14 55 03

File: src/components/HomepageBanner.js
Comment (zstix) [inline src/components/HomepageBanner.js:27]: We might want to leave a comment on what's going on here (or abstract the aspect ratio into a constant at the top of this file).

File: src/components/HomepageBanner.js
Comment (zstix) [inline src/components/HomepageBanner.js:64]: This _might_ solve the issue with the little sliver of background showing through.

```suggestion
              height: calc(100% + 1px);
```

File: src/components/HomepageBanner.js
Comment (zstix) [inline src/components/HomepageBanner.js:23]: If we are positioning all the contents (just one div) with `position: absolute`, then we might not need to set the padding. If we're using the padding for the aspect ratio trick, then nevermind!

File: src/components/HomepageBanner.js
Comment (zstix) [inline src/components/HomepageBanner.js:137]: The theme has a color that's pretty close to the design for this header. We can probably remove this.

```suggestion
```

File: src/pages/index.js
Comment (zstix) [inline src/pages/index.js:307]: I think we want a tiny bit of padding on both the left _and_ the right.
```suggestion
        padding: 0 2rem;
```

**Review by zstix — APPROVED** [review]

**Review by zstix — APPROVED** [review]

**Comment by legnaleama** [comment]
Let's go 🚀!! This looks great, let's take it live! Great work @LizBaker 💪 !

**Comment by austin-schaefer** [comment]
@LizBaker : Pushed up three more commits, basically in order of importance in case any cause issues:

* fd0528a: Must-be-merged, fixes some broken links:
* a4f37f0: Non-fatal but important; tweaks naming and order for consistency with brand guidelines
* 1a9f3e2: Non-fatal, changes link destinations to point at docs rather than index pages

File: src/pages/index.js
Comment (zstix) [inline src/pages/index.js:151]: I ran the query in my local version of GraphQL and I got the following response:

```json
{
  "data": {
    "allMarkdownRemark": {
      "edges": [
        {
          "node": {
            "id": "80d638c5-d1e2-58c3-b4d7-98c3a8581cb2",
            "frontmatter": {
              "title": "Kubernetes Horizontal Pod Autoscaler (HPA) support"
            },
            "fields": {
              "slug": "/whats-new/2021/11/Whats-New-K8sHPA"
            }
          }
        },
```

Which makes me think that, if we're going to type this thing all the way down, we want the types to be something like this (`whatsNewPosts` > `node` and `frontMatter` to `frontmatter`):

```suggestion
      edges: PropTypes.arrayOf(
        PropTypes.shape({
          node: PropTypes.shape({
            frontmatter: PropTypes.shape({
              title: PropTypes.string,
            }),
            fields: PropTypes.shape({
              slug: PropTypes.string,
            }),
          }),
        })
      ),
```

**Comment by LizBaker** [comment]
Waiting on Japanese translations for translations.json before merging

**Review by josephgregoryii — APPROVED** [review]

**Pull request merged**
